src/sv-pipeline/scripts/identify_duplicates.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+
+"""
+Identify and classify duplicated variants from an input VCF.
+"""
+
+from typing import List, Text, Optional
+from collections import defaultdict
+
+import argparse
+import sys
+import pysam
+
+
+def process_duplicates(vcf, fout):
+    # Initialize counters and buffers
+    counts = defaultdict(int)
+    exact_buffer = []
+    ins_buffer = []
+    current_chrom = None
+    current_pos = None
+
+    # Create output files
+    with open(f"{fout}_duplicate_records.tsv", 'w') as f_records, open(f"{fout}_duplicate_counts.tsv", 'w') as f_counts:
+        f_records.write("TYPE\tDUP_RECORDS\n")
+        f_counts.write("TYPE\tDUP_COUNTS\n")
+
+        # Iterate through all records
+        for record in vcf.fetch():
+            # Process current buffers if we've reached a new chrom or pos
+            if record.chrom != current_chrom or record.pos != current_pos:
+                process_buffers(exact_buffer, ins_buffer, counts, f_records)
+                exact_buffer = []
+                ins_buffer = []
+                current_chrom = record.chrom
+                current_pos = record.pos
+
+            # Update buffers with new record
+            exact_key = (
+                record.chrom,
+                record.pos,
+                record.stop,
+                record.info.get('SVTYPE'),
+                record.info.get('SVLEN'),
+                record.info.get('CHR2'),
+                record.info.get('END2'),
+                record.info.get('STRANDS'),
+                record.info.get('CPX_TYPE'),
+                record.info.get('CPX_INTERVALS')
+            )
+            exact_buffer.append((exact_key, record.id))
+
+            if record.info.get('SVTYPE') == 'INS':
+                insert_key = (
+                    record.id,
+                    record.info.get('SVLEN'),
+                    record.alts[0]
+                )
+                ins_buffer.append(insert_key)
+
+        # Process remaining records in the buffer
+        process_buffers(exact_buffer, ins_buffer, counts, f_records)
+
+        # Write counts to file
+        for match_type in sorted(counts.keys()):
+            f_counts.write(f"{match_type}\t{counts[match_type]}\n")
+
+
+def process_buffers(exact_buffer, ins_buffer, counts, f_records):
+    # Process exact matches
+    exact_matches = defaultdict(list)
+    for key, record_id in exact_buffer:
+        exact_matches[key].append(record_id)
+
+    for records in exact_matches.values():
+        if len(records) > 1:
+            counts['Exact'] += 1
+            f_records.write(f"Exact\t{','.join(sorted(records))}\n")
+
+    # Process INS matches
+    for i in range(len(ins_buffer)):
+        for j in range(i + 1, len(ins_buffer)):
+            rec1, rec2 = ins_buffer[i], ins_buffer[j]
+
+            # Size comparison
+            if rec1[1] == rec2[1]:
+                counts['INS 100%'] += 1
+                f_records.write(f"INS 100%\t{rec1[0]},{rec2[0]}\n")
```

Comment:
Let's get rid of whitespace and non-alphanumeric characters for more robust parsing:
```
ins_size_similarity_100
ins_size_similarity_50
ins_size_similarity_0
ins_alt_identical
ins_alt_same_subtype
ins_alt_different_subtype
```

Reply:
Updated accordingly.